src/nupic/regions/tm_region.capnp

```diff
@@ -1,11 +1,17 @@
 @0xb9d11462f08c1dee;
 
+using import "/nupic/algorithms/backtracking_tm.capnp".BacktrackingTMProto;
+using import "/nupic/algorithms/backtracking_tm_cpp.capnp".BacktrackingTMCppProto;
 using import "/nupic/proto/TemporalMemoryProto.capnp".TemporalMemoryProto;
 
-# Next ID: 11
+# Next ID: 13
 struct TMRegionProto {
   temporalImp @0 :Text;
-  temporalMemory @1 :TemporalMemoryProto;
+  union {
+    temporalMemory @1 :TemporalMemoryProto;
+    backtrackingTM @11 :BacktrackingTMProto;
+    backtrackingTMCpp @12 :BacktrackingTMCppProto;
```

Comment:
Field numbers look strange. I think you're trying to leave a gap for existing 2, 3, etc. But do we really need to? Since we're breaking compatibility on the way to NuPIC 1.0, we might as well have straight-forward numbering and not worry about compatibility?